components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/dynamic_prompt.jsp

```diff
@@ -0,0 +1,160 @@
+<%--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+
+<%@ page import="com.google.gson.Gson" %>
+<%@ page import="java.io.ByteArrayOutputStream" %>
+<%@ page import="java.util.Base64" %>
+<%@ page import="java.util.Map" %>
+<%@ page import="java.util.zip.DataFormatException" %>
+<%@ page import="java.util.zip.Inflater" %>
+<%@ page import="org.owasp.encoder.Encode" %>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@include file="localize.jsp" %>
+<%@taglib prefix="e" uri="https://www.owasp.org/index.php/OWASP_Java_Encoder_Project" %>
+<jsp:directive.include file="init-url.jsp"/>
+<jsp:directive.include file="template-mapper.jsp"/>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+
+<%
+    String templateId = request.getParameter("templateId");
+    String promptId = request.getParameter("promptId");
+    String dataStr = request.getParameter("data");
+    Map data = null;
+    String templatePath = null;
+    if (dataStr != null) {
+        byte[] base64DecodedBytes = Base64.getDecoder().decode(dataStr);
```

Comment:
We need to move the deflate/decrypt logic to Java class. JSP should only have styling.